### PR TITLE
Feature/multi trial kfold cv

### PR DESCRIPTION
**Pull request by jasonlaska**
Defines a `RepeatedKFold` cross validation generator and wires in as the default cross-validation scheme.  Adds support for passing cv=(n_folds, n_trials) to `QuicGraphLassoCV` if so desired.

Addresses #59 .   Review @mnarayan ?

### CODE REVIEW

**Comment by mnarayan** [comment]
@jasonlaska 
Just so I understand correctly this, it basically iterates through n_trial*n_folds worth of training/test splits and the actual training/test indices are given by [_iter_test_indices()](https://github.com/skggm/skggm/pull/60/commits/e92723b294a655dc12597d725b62c4a165fbd82e#diff-d8d53f09e3c6beec83e212faebd99be0R79)

So this just modifies KFold's [_iter_test_indices()](https://github.com/scikit-learn/scikit-learn/blob/4d9a12d175a38f2bcb720389ad2213f71a3d7697/sklearn/cross_validation.py#L338)?

The RepeatedKFold looks right. 

Is there any way of knowing which of the cv_scores came from within the folds and which came from the trials before the result gets averaged? Right now I don't think it matters if we can't separate these, but it could come in handy in the future to know which is which. E.g. Label individual cv scores  by their trial number when n_trials>1.  This could come in handy for particular metrics to leverage which error estimates are fully independent and which aren't.

**Comment by jasonlaska** [comment]
- The `RepeatedKFold` is a new, modified implementation of scikit's `KFold`.  The main difference is that it generates `n_trials` shuffles (instead of just one optional one) and then produces k-folds for each shuffled index set.  The actual partitioning is done through the base class (https://github.com/scikit-learn/scikit-learn/blob/4d9a12d175a38f2bcb720389ad2213f71a3d7697/sklearn/cross_validation.py#L65) in the same fashion as `KFold`.

- Wrt to tracking the folds:   The cv generator always spits out results in the same order (shuffle 0 splits 0 to K-1;  shuffle 1 splits 0 to K-1; etc).  The scores are available in this same order (https://github.com/skggm/skggm/blob/feature/multi_trial_kfold_cv/inverse_covariance/quic_graph_lasso.py#L565) and so it should be possible to apply additional analysis to them at that touch point.  My feeling is that without a specific feature to build here, it's premature to build any facility right now.  Current codepath doesn't supports further expansion later if desired.

**Comment by jasonlaska** [comment]
Question:  Right now the only averaging done is for the cv scores.  Was there some mention of actually averaging the covariance/precision estimates too?  The final result produced by `QuicGraphLassoCV` is a refit model (on all data) at the selected `lam_`.

**Comment by mnarayan** [comment]
Nope, only averaging CV scores, not actual precision estimates.
